### PR TITLE
Fix double-marshalling of cachefrom

### PR DESCRIPTION
**Pull request by exFalso**
The API for setting the --cache-from argument is:

```java
BuildImageCmd withCacheFrom(Set<String> cacheFrom);
```
which intuitively indicates that the set of strings passed in should be the imageIds to use for the cache. However this is not the case, and instead users have to "wrap" the arguments in an additional '[]', as illustrated by an existing test `BuildImageCmdIT.cacheFrom`:
```java
        String cacheImage = String.format("[\"%s\"]", imageId1);
        String imageId2 = dockerRule.getClient().buildImageCmd(baseDir2).withCacheFrom(new HashSet<>(Arrays.asList(cacheImage)))
```

As stated in the docker API https://docs.docker.com/engine/api/v1.37/#operation/ImageBuild
the `cachefrom` parameter is a
> JSON array of images used for build cache resolution.

however the current implementation takes the set of strings, where each string itself is an array, and creates a "cachefrom" parameter for each. In other words it seems to require another layer of marshalling.

I *think* the original intention for this API was to take the set of strings where each string is an image id and serialize these into a JSON array automatically. This PR does this.

---
This change is [Reviewable](https://reviewable.io/reviews/docker-java/docker-java/1086)

### CODE REVIEW

File: src/main/java/com/github/dockerjava/jaxrs/BuildImageCmdExec.java [hunk 1]
```diff
@@ -75,10 +76,8 @@ private static AuthConfigurations firstNonNull(final AuthConfigurations fromComm
             webTarget = webTarget.queryParam("t", command.getTag());
         }
 
-        if (command.getCacheFrom() != null) {
-            for (String c: command.getCacheFrom()) {
-               webTarget = webTarget.queryParam("cachefrom", c);
-            }
+        if (command.getCacheFrom() != null && !command.getCacheFrom().isEmpty()) {
+            webTarget = webTarget.queryParam("cachefrom", "[\"" + Joiner.on("\",\"").join(command.getCacheFrom()) + "\"]");
```
Comment (KostyaSha) [inline src/main/java/com/github/dockerjava/jaxrs/BuildImageCmdExec.java]: Is it the same as https://github.com/docker-java/docker-java/blob/master/src/main/java/com/github/dockerjava/core/exec/ListServicesCmdExec.java#L30

Comment (KostyaSha) [reply src/main/java/com/github/dockerjava/jaxrs/BuildImageCmdExec.java]: Please reuse existing helper methods, + parameter should be escaped

Comment (exFalso) [reply src/main/java/com/github/dockerjava/jaxrs/BuildImageCmdExec.java]: @KostyaSha good point! It's not a Map, but I created a similar utility class

**Comment by exFalso** [comment]
@KostyaSha I addressed the comments

**Pull request merged**